### ISSUE TITLE
Pro and Cons for Exit button in Unlock dialog

### ISSUE THREAD

**Issue by igor2x** [pwsafe/pwsafe#1222]
Password Safe v1.19-pre on master branch on Ubuntu 23.10.

1. Click on Lock button from toolbar.
2. File | Unlock to unlock and dialog appears with "Exit" button.

There was some discussion on [PR 1215](https://github.com/pwsafe/pwsafe/pull/1215), but I would like to separate this "Exit" button discussion into separate discussion, to avoid mixing multiple issues into one gigantic fit all never ending discussion.

PRO (retain Exit button):
1. User can quickly click on Exit button to exit program from Unlock dialog and so preserves some time.

CONS (remove Exit button)
1. In my humble opinion this Exit button is very rarely used and removing it would benefit of having simplified Unlock dialog.
2. Argument that Exit button preserves some time is valid, but only to the point if this button is used a by users lot. If not, then users when unlocking database have to go in there brains to digest all of the options. The few they are more time is saved. In my humble opinion Exit button is rarely used and so time spend to digest every time when unlocking is much much bigger then the time preserved when used in rare occasions.
3. It is unusual design with unexpected results. I clicked on this Exit button just once by mistake instead of Cancel button and then wondered why program crashed.
4. It leads to data lost as reported in [issue 1134](https://github.com/pwsafe/pwsafe/issues/1134).
5. If we implement Exit button and are trying to avoid issue 1134, then in order to not lose the data, we are required to ask user if he or she wants to save the data, before exiting. But this is unexpected by the user from Unlock dialog. User wants to exit the program in hurry (like I understand the argument for PRO), but this slow him or her down. And the confusion... at Unlock database dialog we get the prompt of Saving dialog. Sure we can automatically save instead of offering dialog, but this may be undesired by the user and may lead into data lost.
6. Adding Exit button would spare only one click. Instead in some rare occasions, user can just click on Cancel button and then on "X" from top-right to close down the program.
7. Exit button my bring anxiety. When you bring some dangerous actions like Exit button, instead of doing just innocent action like Unlock, there is some dangerous or inconvenient option like Exit button that user is required to start program again if clicked. This brings anxiety to users.
8. Inconsistency between Linux and Windows version of program, because Windows program does not have Exit button.

**Comment by rafaelx** [comment]
Eight good reasons against the exit button and in favor of simplification. Also from the point of view that data integrity is preferable to user-friendliness, in favor of the user.

**Comment by ronys** [comment]
Agree that the Exit button is an unneeded complication, originating in the Windows version, where it can (and should) be removed as well - a separate ticket.
This ticket should be addressed after #1215.

**Comment by igor2x** [comment]
> ronys: _"Windows version ... be removed as well - a separate ticket."_

I have opened separate ticket for Windows in [issue 1224.](https://github.com/pwsafe/pwsafe/issues/1224)

**Comment by igor2x** [comment]
Current issue was fixed by [pull request 1231.](https://github.com/pwsafe/pwsafe/pull/1231). I am closing this issue down.
